src/v/cloud_storage/remote_partition.cc

```diff
@@ -959,73 +959,128 @@ bool remote_partition::bounds_timestamp(model::timestamp t) const {
 static constexpr ss::lowres_clock::duration finalize_timeout = 20s;
 static constexpr ss::lowres_clock::duration finalize_backoff = 1s;
 
-ss::future<> remote_partition::finalize(ss::abort_source& as) {
-    vlog(_ctxlog.info, "Finalizing remote storage state...");
+/// When a remote_partition is being destroyed for the last time, we save this
+/// subset of its state into a background fiber that tries to do a final update
+/// of remote metadata.
+struct finalize_data {
+    model::ntp ntp;
+    model::initial_revision_id revision;
+    cloud_storage_clients::bucket_name bucket;
+    cloud_storage_clients::object_key key;
+    cloud_storage_clients::object_tag_formatter tags;
+    iobuf serialized_manifest;
+    model::offset insync_offset;
+};
+
+ss::future<> finalize_background(remote& api, finalize_data data) {
```

Comment:
Not from this PR, but off the top of my head I didn't have a good answer: we don't do this on read replicas, right?

Reply:
We do not: the check lives in cluster::partition::finalize_remote_partition.

I wanted to double check in this function too, but `remote_partition` doesn't have a reference to the partition's config.

Reply:
Thanks! I always second guess what `is_archival_enabled` actually means, since we still create the archiver on read replicas.